packages/color/src/Color.ts

```diff
@@ -0,0 +1,361 @@
+    toNumber(): number
+    {
+        return this._int;
+    }
+
+    /**
+     * Converts a hexadecimal color number to a string.
+     * @example
+     * import { Color } from 'pixi.js';
+     * new Color('white').toString(); // returns "#ffffff"
+     */
+    toString(): string
+    {
+        const hexString = this._int.toString(16);
+
+        return `#${'000000'.substring(0, 6 - hexString.length) + hexString}`;
+    }
+
+    /**
+     * Convert to a hexidecimal string with alpha.
+     * @example
+     * import { Color } from 'pixi.js';
+     * new Color('white').toHexString(); // returns "#ffffffff"
+     */
+    toHexString(): string
+    {
+        const alphaValue = Math.round(this._components[3] * 255);
+        const alphaString = alphaValue.toString(16);
+
+        return this.toString() + '00'.substring(0, 2 - alphaString.length) + alphaString;
+    }
+
+    /**
+     * Set alpha, suitable for chaining.
+     * @param alpha
+     */
+    setAlpha(alpha: number): this
+    {
+        this._components[3] = alpha;
+
+        return this;
+    }
+
+    /**
+     * Rounds the specified color according to the step.
+     * @param step - Number of steps which will be used as a cap when rounding colors
+     */
+    round(step: number): this
+    {
+        const [r, g, b] = this._components;
+
+        this._components.set([
+            Math.min(255, (r / step) * step),
+            Math.min(255, (g / step) * step),
+            Math.min(255, (b / step) * step),
+        ]);
+
+        return this;
+    }
+
+    /**
+     * Converts color to an [R, G, B, A] array of normalized floats (numbers from 0.0 to 1.0).
+     * @example
+     * import { Color } from 'pixi.js';
+     * new Color('white').toArray(); // returns [1, 1, 1, 1]
+     * @param {number[]|Float32Array} [out] - Output array
+     */
+    toArray<T extends (number[] | Float32Array) = number[]>(out?: T): T
+    {
+        out = out ?? [] as T;
+        const [r, g, b, a] = this._components;
+
+        out[0] = r;
+        out[1] = g;
+        out[2] = b;
+        out[3] = a;
+
+        return out;
+    }
+
+    /**
+     * Normalize the input value into rgba
+     * @param value - Input value
+     */
+    private normalize(value: Exclude<ColorSource, Color>): void
+    {
+        let components: number[];
+
+        if ((Array.isArray(value) || value instanceof Float32Array) && value.length >= 3)
+        {
+            // Handle floats or uints
+            const normalizeValue = (v: number) => (v > 1 ? v / 255 : v);
+
+            components = [
+                normalizeValue(value[0]),
+                normalizeValue(value[1]),
+                normalizeValue(value[2]),
+                value[3] ?? 1.0,
+            ];
+        }
+        else if (value instanceof Uint8Array || value instanceof Uint8ClampedArray)
+        {
+            components = [
+                value[0] / 255,
+                value[1] / 255,
+                value[2] / 255,
+                1.0,
```

Comment:
```suggestion
                value[3] / 255 ?? 1.0,
```

Reply:
If `value[3]` is undefined, then `value[3] / 255` will be `NaN`, and `value[3] / 255 ?? 1.0` will also be `NaN`, instead of `1.0`.

Reply:
What we want here is `(value[3] ?? 255) / 255,`

Reply:
Yes, well seen! 😅